### ISSUE TITLE
False positive when pattern matching single-case DU inside lambda

### ISSUE THREAD

**Issue by rexcfnghk** [fsprojects/FSharpLint#316]
### Description

A `Lint: Match clauses should be in line with 'match' keyword` warning is produced when I pattern-match a single-case DU inside a lambda.

### Repro steps

```fsharp
type Foos = Foos of int []

let foos = Foos [| 1..5 |]
let okFoos = Ok foos
let fooSetResult = Result.map (fun (Foos f) -> Set.ofArray f) // Lint warning on this line
```

### Expected behavior

No warnings produced beccause there is no `match` keyword used.

### Actual behavior

False positive warning.

### Related information 

* Windows 10, VS Code with Ionide-fsharp

**Comment by jgardella** [comment]
Hi @rexcfnghk, thanks for the report.

I chcked on the latest version of master and this doesn't produce an error. I think this was fixed in #310, but Ionide hasn't been updated yet.
